samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -0,0 +1,598 @@
+                    }
+                    # Trim off the completed jobs from our list of jobs
+                    $jobs = $jobs | Where-Object {$_.Id -notin $completedJobs.Id}
+                    # Remove the completed jobs from memory
+                    $completedJobs | Remove-Job
+                }
+                # Wait for 60 sec before checking job status again
+                Start-Sleep -Seconds 60
+            }
+        }
+
+
+        # Needs to create resources in this order, aka parallelize in these three groups, otherwise we get contentions:
+        # i.e. different jobs trying to create the same common resource (RG or lab account)
+        New-ResourceGroups  -ConfigObject $aggregateLabs
+        New-Accounts        -ConfigObject $aggregateLabs
+        New-AzLabMultiple   -ConfigObject $aggregateLabs
+    }
+}
+
+function Set-LabProperty {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Array containing one line for each lab to be created", ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [psobject[]]
+        $labs,
+
+        [Parameter(Mandatory = $true, ValueFromRemainingArguments=$true, HelpMessage = "Series of multiple -propertyName propValue pairs")]
+        [ValidateNotNullOrEmpty()]
+        [string[]]
+        $vars
+    )
+    begin {
+        #Convert vars to hashtable
+        $htvars = @{}
+        $vars | ForEach-Object {
+            if($_ -match '^-') {
+                #New parameter
+                Write-Verbose $_
+                $lastvar = $_ -replace '^-'
+                $lastvar = $lastvar -replace ':' # passing parameters as hashtable inserts a : char
+                $htvars[$lastvar] = $null
+            } else {
+                #Value
+                $htvars[$lastvar] = $_
+            }
+        }
+    }
+
+    process {
+        foreach ($l in $labs) {
+            # Deep cloning not to change the original
+            $lc = [System.Management.Automation.PSSerializer]::Deserialize(
+                    [System.Management.Automation.PSSerializer]::Serialize($l))
+
+            Write-Verbose ($lc | Out-String) 
+
+            function ChangeLab ($lab) {
+                $htvars.Keys | ForEach-Object { $lab.($_) = $htvars[$_]}
+            }
+            $lc | ForEach-Object { ChangeLab  $_}
+            $lc
+        }
+    }
+}
+
+function Show-LabMenu {
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Array containing one line for each lab to be created", ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [psobject[]]
+        $labs,
+
+        [Parameter(Mandatory = $false, HelpMessage = "Pick one lab from the labs' list")]
+        [switch]
+        $PickLab,
+
+        [Parameter(Mandatory = $false, HelpMessage = "Which lab properties to show a prompt for")]
+        [ValidateNotNullOrEmpty()]
+        [string[]]
+        $Properties
```

Comment:
I don't think I understand this and why it would be used.  Looking at the code (and when I ran this locally), when I pass in some property, the property is overridden for all labs.  IE:
".\Cortex.csv" | Import-LabsCsv | Show-LabMenu -Properties "LabName"

This basically overrides the "LabName" with whatever I type in to the command prompt instead of following what the CSV file says.  Is there a real-world (customer) scenario where something like this would be needed?